packages/amplication-build-manager/src/build-runner/build-runner.controller.spec.ts

```diff
@@ -0,0 +1,290 @@
+const spyOnPlainToInstance = jest.spyOn(classTransformer, "plainToInstance");
+const spyOnAxiosPost = jest.spyOn(axios, "post");
+
+describe("BuildRunnerController", () => {
+  let controller: BuildRunnerController;
+  let loggerService: AmplicationLogger;
+  let configService: ConfigService;
+
+  const mockRunnerServiceCopyFromJobToArtifact = jest.fn();
+  const mockRunnerServiceSaveDsgResourceData = jest.fn();
+  const mockKafkaServiceEmitMessage = jest.fn();
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [],
+      controllers: [BuildRunnerController],
+      providers: [
+        {
+          provide: KafkaProducerService,
+          useClass: jest.fn(() => ({
+            emitMessage: mockKafkaServiceEmitMessage,
+          })),
+        },
+        {
+          provide: BuildRunnerService,
+          useClass: jest.fn(() => ({
+            saveDsgResourceData: mockRunnerServiceSaveDsgResourceData,
+            copyFromJobToArtifact: mockRunnerServiceCopyFromJobToArtifact,
+          })),
+        },
+        {
+          provide: ConfigService,
+          useValue: {
+            get: (variable) => {
+              switch (variable) {
+                case Env.CODE_GENERATION_SUCCESS_TOPIC:
+                  return "code_generation_success_topic";
+                case Env.CODE_GENERATION_FAILURE_TOPIC:
+                  return "code_generation_failure_topic";
+                case Env.DSG_RUNNER_URL:
+                  return "http://runner.url/";
+                default:
+                  return "";
+              }
+            },
+          },
+        },
+        MockedAmplicationLoggerProvider,
+      ],
+    }).compile();
+
+    controller = module.get<BuildRunnerController>(BuildRunnerController);
+    configService = module.get<ConfigService>(ConfigService);
+    loggerService = module.get<AmplicationLogger>(AmplicationLogger);
+  });
+
+  it("should be defined", () => {
+    expect(controller).toBeDefined();
+  });
+
+  it("On code generation success, copy file and/or directories from `jobPath` to `artifactPath` and emit Kafka success event", async () => {
+    const codeGenerationSuccessDTOMock: CodeGenerationSuccessDto = {
+      resourceId: "resourceId",
+      buildId: "buildId",
+    };
+    const kafkaSuccessEventMock: CodeGenerationSuccess.KafkaEvent = {
+      key: null,
+      value: { buildId: codeGenerationSuccessDTOMock.buildId },
+    };
+
+    mockRunnerServiceCopyFromJobToArtifact.mockResolvedValue(undefined);
+    mockKafkaServiceEmitMessage.mockResolvedValue(undefined);
+
+    await controller.onCodeGenerationSuccess(codeGenerationSuccessDTOMock);
+
+    expect(mockRunnerServiceCopyFromJobToArtifact).toBeCalledWith(
+      codeGenerationSuccessDTOMock.resourceId,
+      codeGenerationSuccessDTOMock.buildId
+    );
+    await expect(
+      mockRunnerServiceCopyFromJobToArtifact()
+    ).resolves.toBeUndefined();
```

Comment:
@ognjenjevremovic thanks for the contribution! 
One small comment that doesn't require any additional changes unless you want:

Asserting the return type of `void` methods it doesn't really cover any specific business logic and it can be avoided.

Reply:
@overbit thanks for the heads up!  
I totally agree with you; this check does not provide any value overall.

However, I wanted to test if the `Promise` actually `resolves` (and do not `throw`).  
Reading through `Jest` docs, I found a better alternative for this use case:

```js
await expect(
  mockRunnerServiceCopyFromJobToArtifact()
).resolves.not.toThrow()
```

Would you mind me replacing these LOC(s) through a separate commit on this PR? 🙂